## Problem 1.

a) 

Nonrelativistic: Use energy gain from moving across capacitor

$$
\begin{align*}
  &E = T + U
\\&E(x=0) = U = q V
\\&E(x=d) = T = {1 \over 2} m v^2
\\&v = \sqrt{2 q V \over m}
\end{align*}
$$


Relativistic: Use $E = \gamma mc^2$ and $\gamma = {1 \over \sqrt{1 - \beta^2}}$

$$
\begin{align*}
  & q V + mc^2 = \gamma m c^2
\\& {q V\over mc^2} + 1 = \gamma 
% \\ & {q V\over mc^2} + 1 = \sqrt{1 \over 1 - \beta^2}
\\& \sqrt{1 - \beta^2} \left({q V\over mc^2} + 1 \right) = 1
\\& \sqrt{1 - \beta^2} = {1 \over \left({q V\over mc^2} + 1 \right)}
\\& {v^2 \over c^2}= 1-\left( {1 \over \left({q V\over mc^2} + 1 \right)} \right)^2
\\& {v^2}= c^2-\left( {c \over \left({q V\over mc^2} + 1 \right)} \right)^2
\\& v= \sqrt{c^2-\left( {c \over \left({q V\over mc^2} + 1 \right)} \right)^2}

% \\& v^2 = \left( {c \over \left({q V\over mc^2} + 1 \right)} \right)^2 - c^2
% \\& v = \sqrt{\left( {mc^3 \over{q V} + mc^2} \right)^2 - c^2}

\end{align*}
$$
which looks accurate, since the inside of the square root features a term minus the same term over some divisor $>1$.


b) Let's solve numerically



In [32]:
from scipy.constants import m_e, e, c

# m_e kg
# e coulombs
# c m/s
print('nonrelativistic: V=10V and V=10kV (m/s and natural)')
print(f'{(2*e*10/m_e)**(1/2)} {(2*e*10/m_e)**(1/2)/c}')
print(f'{(2*e*10000/m_e)**(1/2)} {(2*e*10000/m_e)**(1/2)/c}')

print('\n')
get_v = lambda V: (c**2 - (c/(e*V/(m_e*c**2)+1))**2)**(1/2)
print('relativistic: V=10V and V=10kV (m/s and natural)')
print(get_v(V=10), get_v(V=10)/c)
print(get_v(V=10000), get_v(V=10000)/c)

nonrelativistic: V=10V and V=10kV (m/s and natural)
1875537.2608284797 0.0062561188941867235
59309695.80731296 0.19783585018443978


relativistic: V=10V and V=10kV (m/s and natural)
1875509.7338270468 0.006256027073993459
58455214.88935105 0.19498560864179929


(c) convert MeV to joules

$$
    E = \gamma mc^2 = T + mc^2
\\  \gamma = {T + mc^2 \over mc^2}
\\  \beta = \sqrt{1 - {1 \over \gamma^2}}
$$

In [37]:
from scipy.constants import m_p
# E = KE + mc^2
E = 100e6 * 1.602e-19 + m_p*c**2 # joules
gamma = E/(m_p*c**2)
beta = (1-1/(gamma**2))**(1/2)
print('gamma, beta, m/s')
print(gamma, beta, beta*c)

gamma, beta, m/s
1.1065671424093946 0.42817523295658855 128363705.54277828


d) Protons and neutrons have the same mass for our purposes. Since kinetic energy scales linearly with mass, this allows for rescaling $\gamma$ as

$$
    \gamma = {T + n\times m_p c^2 \over n \times m_pc^2}
\\  = {n{T \over n} + n\times m_p c^2 \over n \times m_pc^2}
\\  = {{T \over n} + m_p c^2 \over m_pc^2}
$$
And $\beta$ is fixed the same way as $\gamma$. Then all heavy nuclei $\beta$ are scaled equivalently by kinetic energy over atomic mass number. Since this is equivalent for all heavy nuclei, specifying MeV/nucleon of an RF cavity structure always describes the energy of the objects it accelerates.

<div style="page-break-after: always">  </div>

## Problem 2.

(a) We need to compute $\gamma$, let's convert from total energy per nucleon to total energy

$$
\begin{gathered}
    E = \gamma m c^2
\\  m = 197 \times 0.93113 \text{ GeV / c}^2 = A \times m_p
\\  E = A {E \over A}
\\  A = 197
\\  \gamma = {A \times {E \over A} \over m c^2}
\\  = {{E \over A} \over m_p}
\end{gathered}
$$

In [ ]:
E_inj = 10.5
E_col =  100
m_p = 0.93113 # avg gold nucleon energy

gamma_inj = E_inj / m_p
gamma_col = E_col / m_p

v_inj = (1 - (1/gamma_inj**2))**(1/2) * c
v_col = (1 - (1/gamma_col**2))**(1/2) * c

# print(v_inj/c, v_col/c)
circumference = 3834
f1 = v_inj/circumference
f2 = v_col/circumference
print('Injection, Collision, Difference (Hz)')
print(f1, f2, f2-f1)

Injection, Collision, Difference (Hz)
77885.06817905072 78189.73961114282 304.6714320921019


(b) From class we have:

$$
\theta = - {\ell \over \rho} = - \ell \left( q B_y \over p \right) = - {\Delta x \over f}
$$

Each magnet is 10m, and must bend by $\theta = {2 \pi \over 192} \sim 0.03$ radians. We'll drop all signs for now, which yields

$$
\begin{gathered}
    B_y = {\theta \over \ell} {p \over q}
\\  B_y = {\theta \over \ell} {\gamma m v \over q}
\end{gathered}
$$

In [70]:
from scipy.constants import pi

theta = 2*pi / 192 # rad
ell = 10 # m
m = 197 * 0.93113 * 1.782662e-27 # kg
q = 79 * e # C

B = theta/ell * gamma_col * m * v_col / q
print(B)

2.721938216061688


So we have $B = 2.7$ T

<div style="page-break-after: always">  </div>

## Problem 3.

(a) We'll use the chain rule. Let's start by laying out some relevant equations

<!-- $$
\begin{gathered}
        \vec r 
    =   (\rho + x) \hat x + z \hat z
    % =   \rho \hat x + x \hat x + z \hat z
\\      \dot{\hat z} 
    =   0
\\      
\end{gathered}
$$ -->

$$
\begin{align*}
        {d \vec r \over d t} 
  & =   {d \over dt} [\rho \hat x]
    +   {d \over dt} [x \hat x]
    +   {d \over dt} [z \hat z]
\\& =   \cancel{\dot \rho \hat x}
    +   \rho \dot{\hat x}
    +   \dot x \hat x
    +   x \dot{\hat x}
    +   \dot z \hat z
    +   \cancel{z \dot{ \hat z}}
\\& =   \dot x \hat x
    +   (\rho + x) \dot{\hat x}
    +   \dot z \hat z
\end{align*}
$$
Let's work on $\hat x$
$$
        {d \hat x \over dt} 
    =   {d \hat x \over d s} {ds \over dt}
\\[0.5em]  
    =   {1 \over \rho} \hat s {ds \over d \theta} {d \theta \over dt}
\\[0.5em]  
    =   {1 \over \rho} \dot \theta \hat s {ds \over d \theta}
    =   \dot \theta \hat s
\\      \dot{\vec r}
    =   \dot x \hat x
    +   (\rho + x) \dot \theta \hat s
    +   \dot z \hat z

$$

<!-- On the RHS, we can expand as
$$
\begin{gathered}
        \dot \theta 
    =   {d \over dt} \left[ s \over \rho \right]
    =   {\rho \dot s - \dot \rho s \over \rho^2}
        
\\      (\rho + x) \dot \theta \hat s
    =   (\rho + x) {\rho \dot s - \dot \rho s \over \rho^2} \hat s

\end{gathered} 
$$ -->

Next up: $\ddot{\vec r}$.

$$
\begin{align*}
        {d \dot{\vec r} \over dt}
   &={d \over dt}\left[\dot x \hat x                 \right]
    +{d \over dt}\left[(\rho + x) \dot \theta \hat s\right]
    +{d \over dt}\left[\dot z \hat z                 \right]
\\ &=\ddot{x} \hat x
    +\dot{x}  \dot{\theta} \hat s
    +\dot \theta \hat s{d \over dt}\left[\rho + x\right]
    +(\rho + x) {d \over dt}\left[\dot \theta \hat s \right]
    +\ddot z \hat z
\\ &=\ddot{x} \hat x
    +2\dot{x}  \dot{\theta} \hat s
    +(\rho + x) \ddot \theta \hat s 
    +(\rho + x) \dot \theta {d \over dt}\hat s 
    +\ddot z \hat z
\\ &=\ddot{x} \hat x
    +2\dot{x}  \dot{\theta} \hat s
    +(\rho + x) \ddot \theta \hat s 
    +(\rho + x) \dot \theta {d \over dt}\hat s 
    +\ddot z \hat z

\end{align*}
$$

The next step is to get a $\hat x$ out of $\dot{\hat s}$ 

$$
    {d \hat s \over d t} = {d \hat s \over d s} {d s \over dt} = - {\rho \over \rho} \hat x {d \theta d t}
\\      \ddot{\vec r} 
    =   \left[ \ddot x - (\rho + x) \dot{\theta}^2\right] 
    +   \left[ 2 \dot x \dot \theta + (\rho + x)\ddot \theta \right] \hat s + \ddot z \hat z
$$

(b) Note that $\vec v = \dot{\vec r}$. Since our unit vectors are orthogonal, we can use the typical cross product determinant. Note that $\dot{\gamma} = 0$ in a static magnetic field.

$$
\begin{gathered}
e\dot{\vec r} \times \vec B = e\begin{vmatrix} 
    \hat x  & \hat s                & \hat z
\\  \dot x  & (\rho + x)\dot \theta & \dot z
\\  B_x     & 0                     & B_z
\end{vmatrix}
\\  = e(\rho+x) \dot \theta B_z  \hat x  
    + e(\dot z B_x -  \dot x B_z)\hat s 
    - e(\rho + x)\dot \theta B_x \hat z
% \\      {d \vec p \over dt} 
%     =   \gamma m \ddot{\vec r}
%     =   
\end{gathered}
$$

Pairing with $\gamma m \ddot{\vec r}$ we have

$$
\begin{gathered}
        \gamma m\left[\ddot x - (\rho+x)\dot \theta^2 \right]
    =   e(\rho + x)\dot \theta B_z
% \\      2 \dot x
\\      \gamma m\ddot z
    =   -e(\rho + x) \dot \theta B_x
\end{gathered}
$$

Comparing $v_s$ with $\dot s$, we observe $v_s = (\rho + x)\dot \theta$.

$$
    v_s dt = (\rho + x) d \theta 
\quad \rightarrow \quad
    v_s = (\rho + x) {d\theta \over dt}
    %     B\rho
    % =   {\gamma m \rho \dot \theta}
\\      \ddot z 
    = - {e \over \gamma m}{(\rho + x)\dot \theta B_x}
    = - {v_s^2 B_x \over B\rho}
\\      \ddot x - (\rho + x)\dot \theta^2
    =   {v_s^2 B_z \over B\rho}
$$


(c) We'll start with the shorter formula

$$
        {d^2 z \over dt^2}
    =   {d^2 z \over ds^2} \left(d s \over dt \right)^2 
    + 0
\\      z''
    =   - {B_x \over B\rho} {v_s^2 \over \dot s^2}
    =   - {B_x \over B\rho} \left(1 + {x \over \rho}\right)^2
$$
Where the final step uses $v_s / \dot s = (\rho + x)\dot \theta / \rho \dot \theta = 1 + x/\rho$.

$$
        x'' - (\rho + x)\dot \theta^2 {1 \over \dot s^2}
    =   {B_z \over B\rho} \left(1 + {x \over \rho}\right)^2
$$
Here, we have

$$
(\rho + x)\dot \theta^2 {1 \over \dot s^2}
=   {v_s \dot \theta \over \dot s^2}
=   \left(1 + {x \over \rho}\right) {1 \over \rho}
=   {\rho + x \over \rho^2}
$$
yielding the final expression
$$
        x'' - {\rho + x \over \rho^2}
    =   {B_z \over B\rho} \left(1 + {x \over \rho}\right)^2

$$
